### PR TITLE
[DiscordCoreAPI] Updating to version 2012-09-25

### PR DESCRIPTION
**Pull request by RealTimeChris**
**Describe the pull request**

- #### What does your PR fix?
  I've switched out nlohmann-json for simdjson - a much more rapid library for parsing json, as well as implemented my own 
  Json serializer, along with a ring buffer for incoming data - all of which has significantly increased the rate at which this library 
  is capable of parsing incoming json data.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <Linux x64, Windows x64>, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  I believe that it does.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes I have.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: versions/d-/discordcoreapi.json [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+      "port-version": 0
+    },
+    {
+      "git-tree": "20aede4355046a37c4494501f1960d58ed75a807",
       "version-date": "2022-08-07",
       "port-version": 0
     },
     {
-      "git-tree": "81b6a00219f65e2ccabcf1330b132f1c2fdbc361",
+      "git-tree": "d3519d6dd1dc6918b16f9c9679c23a12ae673456",
```
Comment (JonLiu1993) [inline versions/d-/discordcoreapi.json:14]:
```suggestion
      "git-tree": "81b6a00219f65e2ccabcf1330b132f1c2fdbc361",
```

File: versions/d-/discordcoreapi.json [hunk 2]
```diff
@@ -1,12 +1,17 @@
 {
   "versions": [
     {
-      "git-tree": "d3519d6dd1dc6918b16f9c9679c23a12ae673456",
+      "git-tree": "0d2e0511274658095bc45584a95d7964017f0f9a",
+      "version-date": "2022-09-25",
+      "port-version": 0
+    },
+    {
+      "git-tree": "20aede4355046a37c4494501f1960d58ed75a807",
```
Comment (JonLiu1993) [inline versions/d-/discordcoreapi.json:9]:
```suggestion
      "git-tree": "d3519d6dd1dc6918b16f9c9679c23a12ae673456",
```

Comment (JonLiu1993) [reply versions/d-/discordcoreapi.json:9]: @RealTimeChris ,Thanks for your pr, please revert the history `git-tree`, we need to ensure that the historical git-tree remains unchanged to facilitate backtracking to its historical version.

File: versions/d-/discordcoreapi.json (hunk 1 above)
Comment (JonLiu1993) [reply versions/d-/discordcoreapi.json:14]: Same as above

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**